tests/test_models.py

```diff
@@ -0,0 +1,1194 @@
+              - name: test-name
+                channel: test-channel
+            """
+        ),
+    )
+
+    with pytest.raises(CraftError, match="field 'type' required in top-level configuration"):
+        load(tmp_path)
+
+
+def test_load_minimal_metadata_from_charmcraft_yaml_missing_summary(
+    tmp_path, prepare_charmcraft_yaml
+):
+    """Load a mimimal charmcraft.yaml with metadata. But missing summary."""
+    prepare_charmcraft_yaml(
+        dedent(
+            """
+            name: test-charm-name
+            type: charm
+            description: test-description
+
+            bases:
+              - name: test-name
+                channel: test-channel
+            """
+        ),
+    )
+
+    with pytest.raises(CraftError, match="needs value in field 'summary'"):
+        load(tmp_path)
+
+
+def test_load_minimal_metadata_from_charmcraft_yaml_missing_description(
+    tmp_path, prepare_charmcraft_yaml
+):
+    """Load a mimimal charmcraft.yaml with metadata. But missing description."""
+    prepare_charmcraft_yaml(
+        dedent(
+            """
+            name: test-charm-name
+            type: charm
+            summary: test-summary
+
+            bases:
+              - name: test-name
+                channel: test-channel
+            """
+        ),
+    )
+
+    with pytest.raises(CraftError, match="needs value in field 'description'"):
+        load(tmp_path)
+
+
+def test_load_minimal_metadata_from_metadata_yaml(
+    tmp_path, prepare_charmcraft_yaml, prepare_metadata_yaml
+):
+    """Load a mimimal charmcraft.yaml with full metadata. (Spec ST087)"""
+    prepare_charmcraft_yaml(
+        dedent(
+            """
+            type: charm
+
+            bases:
+              - name: test-name
+                channel: test-channel
+            """
+        ),
+    )
+    prepare_metadata_yaml(
+        dedent(
+            """
+            name: test-charm-name
+            summary: test-summary
+            description: test-description
+            """
+        ),
+    )
+
+    config = load(tmp_path)
+
+    assert config.name == "test-charm-name"
+    assert config.type == "charm"
+    assert config.summary == "test-summary"
+    assert config.description == "test-description"
+    assert config.bases == [
+        BasesConfiguration(
+            **{
+                "build-on": [
+                    Base(
+                        name="test-name",
+                        channel="test-channel",
+                        architectures=[get_host_architecture()],
+                    )
+                ],
+                "run-on": [
+                    Base(
+                        name="test-name",
+                        channel="test-channel",
+                        architectures=[get_host_architecture()],
+                    )
+                ],
+            }
+        )
+    ]
+    assert config.metadata_legacy
+
+
+def test_load_minimal_metadata_from_metadata_yaml_missing_name(
```

Comment:
These could be parametrized, but I'm not going to hold up the PR on that — we can do it another time.